storage/shard.go

```diff
@@ -673,16 +744,13 @@ func (s *dbShard) insertSeriesEntries(inserts []dbShardInsert) error {
 		// Insert still pending, perform the insert
 		entry = inserts[i].entry
 		s.lookup[entry.series.ID().Hash()] = s.list.PushBack(entry)
-
-		// If we are going to write to this entry then increment the
-		// writer count so it does not look empty immediately after
-		// we release the write lock
-		if inserts[i].hasPendingWrite {
-			entry.incrementWriterCount()
-		}
 	}
 	s.Unlock()
 
+	if !anyPendingWrites {
+		return nil
+	}
+
```

Comment:
I know this isn't in your changes, but should this method capture and return the errors from `entry.series.Write` in the loop below this line?